examples/step-37/doc/results.dox

```diff
@@ -227,13 +292,120 @@ elements increases (the matrix-free implementation has costs
+
+The following figure shows a scaling experiments on $\mathcal Q_3$
+elements. Along the lines, the problem size is held constant as the number of
+cores is increasing. When doubling the number of cores, one expects a halving
+of the computational time, indicated by the dotted gray lines. The results
+show that the implementation shows almost ideal behavior until an absolute
+time of around 0.1 seconds is reached. The solver tolerances have been set
+such that the solver performs five iterations. This way of plotting data is
+the <b>strong scaling</b> of the algorithm. As we go to very large core
+counts, the curves flatten out a bit earlier, which is because of the
+communication network in SuperMUC where communication between processors
+farther away is slightly slower.
+
+<img src="https://www.dealii.org/images/steps/developer/step-37.scaling_strong.png" alt="">
+
+In addition, the plot also contains results for <b>weak scaling</b> that lists
+how the algorithm behaves as both the number of processor cores and elements
+is increased at the same pace. In this situation, we expect that the compute
+time remains constant. Algorithmically, the number of CG iterations is
+constant at 5, so we are good from that end. The lines in the plot are
+arranged such that the top left point in each data series represents the same
+size per processor, namely 131,072 elements (or approximately 3.5 million
+degrees of freedom per core). The gray lines indicating ideal strong scaling
+are by the same factor of 8 apart. The results show again that the scaling is
+almost ideal. The parallel efficiency when going from 288 cores to 147,456
+cores is at around 75% for a local problem size of 750,000 degrees of freedom
+per core which takes 1.0s on 288 cores, 1.03s on 2304 cores, 1.19s on 18k
+cores, and 1.35s on 147k cores. The algorithms also reach a very high
+utilization of the processor. The largest computation on 147k cores reaches
+around 1.7 PFLOPs/s on SuperMUC out of an arithmetic peak of 3.2 PFLOPs/s. For
+an iterative PDE solver, this is a very high number and significantly more is
+often only reached for dense linear algebra. Sparse linear algebra is limited
+to a tenth of this value.
+
+As mentioned in the introduction, the matrix-free method reduces the memory
+consumption of the data structures. Besides the higher performance due to less
+memory transfer, the algorithms also allow for very large problems to fit into
+memory. The figure below we show the computational time as we increase the
+problem size until an upper limit where the computation exhausts memory. We do
+this for 1k cores, 8k cores, and 65k cores and see that the problem size can
+be varied over almost two orders of magnitude with ideal scaling. The largest
+computation shown in this picture involves 292 billion ($2.92 \cdot 10^{11}$)
+degrees of freedom. On a DG computation of 147k DoFs, the above algorithms
+were also run on 532 billion DoFs.
```

Comment:
` of 147k DoFs`->`with 147k cores` and `on 532 billion DoFs`->`for 532 billion DoFs`?